### PR TITLE
WT-9588 Turn a check into assert in __rec_fill_tw_from_upd_select

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

**Review by kommiharibabu — APPROVED** [review]
LGTM

File: src/reconcile/rec_visibility.c [hunk 1]
```diff
@@ -657,50 +657,45 @@ __rec_fill_tw_from_upd_select(
...
-         * when it is written to the disk image in the previous eviction.
+         * If the tombstone is restored from the disk or the history store, the onpage value and the
+         * history store value should have been restored together. Therefore, we should not end up
+         * here.
          */
-        if (!F_ISSET(tombstone, WT_UPDATE_RESTORED_FROM_DS | WT_UPDATE_RESTORED_FROM_HS)) {
-            WT_RET(__rec_append_orig_value(session, page, tombstone, vpack));
+        WT_ASSERT_ALWAYS(session,
+          !F_ISSET(tombstone, WT_UPDATE_RESTORED_FROM_DS | WT_UPDATE_RESTORED_FROM_HS),
+          "Tombstone have been written to the disk image or history store should be accompanied by "
```
Comment (korteland) [inline src/reconcile/rec_visibility.c:666]: Grammar nitpick: "Tombstone have been written" should be "A tombstone written"

**Review by korteland — APPROVED** [review]
LGTM! (Grammar fix is optional.)

File: src/reconcile/rec_visibility.c (hunk 1 above)
Comment (quchenhao) [reply src/reconcile/rec_visibility.c:666]: Thanks

**Comment by quchenhao** [comment]
evergreen merge

**Pull request merged**